test/src/test/java/org/corfudb/runtime/collections/CorfuQueueTest.java

```diff
@@ -105,9 +117,133 @@ public String toString() {
         bmap.put(new ByteArray("fg".getBytes()), "fg");
         bmap.put(new ByteArray("abcd".getBytes()), "abcd");
         for (Map.Entry<ByteArray, String> b : bmap.entrySet()) {
-            System.out.println(b);
+            log.debug("Entry {}", b);
         }
+    }
+
+    @Test
+    public void queueBackwardsCompatibility() {
+        CorfuQueue oldQueueInstance1 = new CorfuQueue(getDefaultRuntime(), "test",
+                Serializers.JSON);
+
+        CorfuRuntime rt2 = getNewRuntime(getDefaultNode()).connect();
+        CorfuQueue oldQueueInstance2 = new CorfuQueue(rt2, "test",
+                Serializers.QUEUE_SERIALIZER);
+
+        // produce items to the queue with two different serializers from two different clients
+        final int numItemsToProduce = 10;
+        IntStream.range(0, numItemsToProduce).forEach(itemIdx -> {
+            if (itemIdx % 2 == 0) {
+                oldQueueInstance1.enqueue(getByteString(String.valueOf(itemIdx)));
```

Comment:
just wondering for code coverage of the sequence and id part can this test case wrap the enqueue in a transaction?

Reply:
I think there's a bug. When I wrap oldQueueInstance1 with a transaction, but not oldQueueInstance2, then the read order is flipped. If there is no transaction, then how is addPreCommitListener called?

Reply:
it won't be called - we can't support a mixed case like that anyway

Reply:
Why not?

If it's not allowed then the operation should be failed.

Reply:
Also, most of the tests don't use transactions to produce to the queue, so I'm not sure why this isnt allowed.

Reply:
Even if we don't mix transactional and non-transaction. Just having non-transactional enqueues will produce an incorrect order.

Reply:
so you suggest we just throw error if queue is used outside a transaction?

Reply:
Yea, it won't make sense to allow it if its incorrect. #2712